qrgui/mainWindow/scriptAPI/guiFacade.cpp

```diff
@@ -128,3 +135,55 @@ QTreeWidgetItem *GuiFacade::propertyTreeWidgetItem(const QString &name) const
 
 	return nullptr;
 }
+
+QWidget *GuiFacade::deepViewPort(const QObject *object) const
+{
+	return object->findChild<QWidget *>("qt_scrollarea_viewport");
+}
+
+QWidget *GuiFacade::viewPort(const QAbstractScrollArea *widget) const
+{
+	return widget->viewport();
+}
+
+QMenu *GuiFacade::getMenu(const QString &menuName) const
+{
+	QMenuBar *menuBar = mMainWindow.findChild<QMenuBar *>();
+	QMenu *menu = menuBar->findChild<QMenu *>(menuName, Qt::FindDirectChildrenOnly);
+	return menu;
+}
+
+QAction *GuiFacade::getActionInMenu(const QMenu *menu, const QString &actionName) const
+{
+	QList<QAction *> actions = menu->actions();
+	for (QAction * const action: actions) {
+		if (action->objectName() == actionName || action->text() == actionName) {
+			return action;
+		}
+	}
+
+	return nullptr;
+}
+
+bool GuiFacade::isSubMenuInMenu(const QMenu *menu, const QAction *action) const
+{
+	return menu->children().contains(action->menu());
+}
+
+QMenu *GuiFacade::getMenuContainedByAction(QAction *action) const
+{
+	emit action->menu()->aboutToShow();
```

Comment:
+	/// Need repeatedly to call this method (if the menu is opening again) because
 +	/// this method emits necessary signal (see warning).
 +	/// @warning it needs because the appropriative bug exists: https://bugs.launchpad.net/appmenu-qt5/+bug/1449373.